src/components/ui/combobox.tsx

```diff
@@ -113,25 +113,23 @@ export function ComboBox({
 
   const handleSelect = (clickedOption: DefaultValueOption): void => {
     if (max) {
-      if (max < 0 || max > options.length) {
-        max = 0
+      if (max <= 0 || max >= options.length) {
+        max = options.length
       }
     }
     const clickedIsAlreadySelected = selected.find(selOption => selOption.value === clickedOption.value)
     let newOptions
-    if (max > 0 && !clickedIsAlreadySelected && max && selected.length >= max) {
+    if (clickedIsAlreadySelected) {
+      newOptions = [...selected].filter(selOption => selOption.label !== clickedOption.label)
```

Comment:
Not that it matters much. But it is a bit unnecessary to use the spread syntax to create a new shallow copy of the array as Array.filter() also creates a new array with the shallow copied elements.

Reply:
Of course, will fix asap